### ISSUE TITLE
Downloading from github

### ISSUE THREAD

**Issue by dostalradim** [jjangga0214/hasura-cli#27]
Hi, when I install hasura-cli with npm, the package is trying to download binary from github. But my servers are behind proxy, how can I tell the hasura-cli package to use my http/s proxy? I put proxy setting in npm config but it did not help.

Thank you

**Comment by jjangga0214** [comment]
@dostalradim 
Hi! Thanks for making an issue!
I guess maybe this issue is orthogonal(out of package-level responsibility?) to the npm package.
I am not sure if the feature(configuring alternative download url) is generally what an individual package should provide. 

Would you find a way to just simply configure your proxy to handle an outbound GET request to `https://github.com/hasura/graphql-engine/releases/download/${tag}/${asset}` (e.g. `https://github.com/hasura/graphql-engine/releases/download/v1.1.1/cli-hasura-linux-amd64`) to `https://url-you-want`?

**Comment by dostalradim** [comment]
Yes, it seems better to me because npm developers cannot force proxy to your code :). And is possible to get proxy form npm configuration? Or maybe from environment HTTP_PROXY and HTTPS_PROXY variables instead create new one?

Thank you!

**Comment by jjangga0214** [comment]
@dostalradim 

I just want to check one thing to make it clear.

Isn't it the best practice to apply a proxy "automatically" on OS or container level, rather than package level? 

I don't know how people are doing, so curious about it.

**Comment by dostalradim** [comment]
Yes, that mentioned variables are used for this. I think that using these two environmental variables is best. Same like curl application for example, it will use them when they exists too. But I am not expert of npm packaging.

**Comment by honeycruse** [comment]
I have same problem, `npm` use environment variables `http_proxy` and `https_proxy`, but the function [asset.ts/download](https://github.com/jjangga0214/hasura-cli/blob/master/src/asset.ts) doesn't use them.
I use `proxychains` to solve this proxy problem temporarily.

**Comment by jjangga0214** [comment]
PR is welcome!
